src/main/java/io/kokuwa/maven/helm/util/DependencyOverwriter.java

```diff
@@ -0,0 +1,233 @@
+			if (line.contains("dependencies:")) {
+				isDependencies = true;
+				overwrittenChart.add(line);
+				continue;
+			}
+
+			//check for lines that need to be overwritten
+			if (isDependencies != null && isDependencies) {
+				if (line.contains("version:")) { 
+					version = true;
+					
+					//if no overwrite value is provided for the version, the original will be used
+					if (currentDependency.isOverwrite() && overwriteVersion != null && !overwriteVersion.isEmpty()) {
+						log.info("Overwriting dependency '" + currentDependency.getName() + "' with new version:");
+						log.info("\tOld Value:");
+						log.info(line);
+
+						//preserve any characters/spacing that may appear around the attribute
+						line = line.replace(currentDependency.getVersion(), overwriteVersion);
+
+						log.info("\tNew Value:");
+						log.info(line);
+					}
+				} else if (line.contains("repository:")) {
+					repository = true;
+
+					if (currentDependency.isOverwrite()) {
+						log.info("Overwriting dependency '" + currentDependency.getName() + "' with new repository:");
+						log.info("\tOld Value:");
+						log.info(line);
+
+						//preserve any characters/spacing that may appear around the attribute
+						line = line.replace(currentDependency.getRepository(), overwriteRepository);
+
+						log.info("\tNew Value:");
+						log.info(line);
+					}
+				} else if (line.contains("name:")) { 
+						name = true;
+				}
+
+				//Repository is not a required field so could be null
+				if (currentDependency.getRepository() == null) {
+					repository = true;
+				}
+			}
+
+			overwrittenChart.add(line);
+
+			//check if the current dependency has been read completely
+			if (name && version && repository) {
+				name = version = repository = false;
+
+				if (dependencyIterator.hasNext()) {
+					currentDependency = dependencyIterator.next();
+				} else {
+					isDependencies = false;
+				}
+			}
+		}
+
+		//overwrite the existing Chart.yaml with the new values
+		try {
+			File newFile;
+			if (this.writeDirectory != null) {
+				newFile = new File(this.writeDirectory.toString() + "/Chart.yaml");
+			} else {
+				newFile = new File(directory.toString() + "/Chart.yaml");
+			}
+			Files.write(Paths.get(newFile.toURI()), overwrittenChart, StandardCharsets.UTF_8);
+			log.info("Overwriting successful for " + newFile);
+		} catch (IOException e) {
+			throw new MojoExecutionException("Unable to overwrite Chart.yaml at " + directory, e);
+		}
+	}
+
+	/**
+	 * Utility class for overwriting local chart dependencies when making use of helm.overwriteLocalDependencies
+	 * @param overwriteRepository used to overwrite a local path chart's repository
+	 * @param overwriteVersion used to overwrite a local path chart's version
+	 * @param directory
+	 * @param log
+	 * @throws MojoExecutionException
+	 */
+	public void execute(String overwriteRepository, String overwriteVersion, Path directory, Log log) 
```

Comment:
A: Since `overwriteRepository`, `overwriteVersion`, and `log` are all constant across calls to this method, it might make sense to move them to the constructor.